native-sql-engine/core/src/main/scala/com/intel/oap/execution/ColumnarShuffledHashJoinExec.scala

```diff
@@ -128,6 +128,18 @@ case class ColumnarShuffledHashJoinExec(
       case _ =>
         throw new UnsupportedOperationException(s"Join Type ${joinType} is not supported yet.")
     }
+    // build check for leftKeys and rightKeys
+    leftKeys.union(rightKeys).foreach { expr =>
+      val keyExpr =
+        ColumnarExpressionConverter.replaceWithColumnarExpression(expr)
+      val supportCodegen =
+        keyExpr.asInstanceOf[ColumnarExpression].supportColumnarCodegen(null)
+      this.supportCodegen = this.supportCodegen && supportCodegen
+      if (!supportCodegen) {
```

Comment:
Thanks for your fix!
Currently, there is only codegen impl. for SHJ with condition (see the first three lines in `#doExecuteColumnar`). So only for condition case, if the key expression is not supporting codegen, we should let the operator fall back. For no condition case, we just need to set `this.supportCodegen` flag to avoid the attempt of going into codegen, as you already did in line 137.

So my suggestion is removing the logic of throwing exception here. And change line 151 to `if (!this.supportCodegen)` to just let it fall back in condition case.

Reply:
Actually, condition is None, it won't run into line 151 with no condition case.

Reply:
Hi @jackylee-ch, for no condition case, I think non-codegen impl. can work well in your case and it looks no need to fall back. Could you please have a test after removing throwing `UnsupportedOperationException` here to allow it go into non-codegen impl?

Reply:
Oh, got it, thanks

Reply:
I have tested it in our environment, it works. Thanks!